core/src/main/java/com/linecorp/armeria/internal/common/RequestContextUtil.java

```diff
@@ -48,6 +57,61 @@
     private static final Set<Thread> REPORTED_THREADS =
             Collections.newSetFromMap(new MapMaker().weakKeys().makeMap());
 
+    private static final RequestContextStorage requestContextStorage;
+
+    static {
+        final List<RequestContextStorageProvider> providers = ImmutableList.copyOf(
+                ServiceLoader.load(RequestContextStorageProvider.class));
+        final String providerFqcn = Flags.requestContextStorageProvider();
+        if (!providers.isEmpty()) {
+
+            RequestContextStorageProvider provider = null;
+            if (providers.size() > 1) {
+                if (providerFqcn.isEmpty()) {
+                    throw new IllegalStateException(
+                            "Found more than one " + RequestContextStorageProvider.class.getSimpleName() +
+                            ". You must specify -Dcom.linecorp.armeria.requestContextStorageProvider=<FQCN>." +
+                            " providers: " + providers);
+                }
+
+                for (RequestContextStorageProvider candidate : providers) {
+                    if (candidate.getClass().getName().equals(providerFqcn)) {
+                        if (provider != null) {
+                            throw new IllegalStateException(
+                                    providerFqcn + " matches more than one " +
+                                    RequestContextStorageProvider.class.getSimpleName() + ". providers: " +
+                                    providers);
+                        } else {
+                            provider = candidate;
+                        }
+                    }
+                }
+                if (provider == null) {
+                    throw new IllegalStateException(
+                            providerFqcn + " does not match any " +
+                            RequestContextStorageProvider.class.getSimpleName() + ". providers: " + providers);
+                }
+            } else {
+                provider = providers.get(0);
+            }
+
+            try {
+                requestContextStorage = provider.newStorage();
+            } catch (Throwable t) {
+                throw new IllegalStateException("Failed to create context storage. provider: " + provider, t);
+            }
+            logger.info("{} is used to create the request context storage: {}",
```

Comment:
I actually meant not logging when:

- using the default provider (because it may be noisy.)
- specified via the flag (because it's logged there already.)

i.e. we can log in the `else {}` block above.

Reply:
I thought it might be useful to show the information of the created request context storage after it's fully created. (i.e. no exception when calling `newStorage()`).

Reply:
But I guess showing the created storage context is too verbose. Fixed. Thank you!